### ISSUE TITLE
IOB Calculation Does Not Use Proper Insulin Duration (Humalog)

### ISSUE THREAD

**Issue by adanvdo** [NightscoutFoundation/xDrip#1772]
Regardless of my prediction settings or insulin profile, xdrip appears to be using an insulin duration of ~2.4 hours to calculate IOB.

I have tried with "multiple insulin profiles" both enabled and disabled.  My low level insulin duration setting is at 3 hours.

Compared to the omnipod, xdrip is reporting a much lower IOB which results in a higher suggested bolus.  Here is a specific example.

2:27 PM - Bolus 1 unit of Humalog.
4:26 PM - Omnipod shows IOB of 0.4 and suggests 0.5 units, but xdrip shows IOB of 0.27 and suggests 1 unit.

I usually try to bolus based on xdrip's suggestion, but for some reason there is a big difference between xdrip and omnipod.

![IMG_20210702_162223](https://user-images.githubusercontent.com/26498008/124325426-934c8980-db52-11eb-92b0-fc2639e5b66b.jpg)
![IMG_20210702_162209](https://user-images.githubusercontent.com/26498008/124325428-947db680-db52-11eb-9f02-d19582f3782b.jpg)
![Screenshot_20210702-161925](https://user-images.githubusercontent.com/26498008/124325430-95164d00-db52-11eb-8882-8ca5965e5dbe.jpg)
![Screenshot_20210702-161919](https://user-images.githubusercontent.com/26498008/124325431-95164d00-db52-11eb-88c1-930c0dffafa4.jpg)
![Screenshot_20210702-161954](https://user-images.githubusercontent.com/26498008/124325432-95164d00-db52-11eb-9aca-d1476640e41e.jpg)

**Comment by BenoitAnastay** [comment]
Maybe check this

https://albertadiabeteslink.ca/process.php?docname=Understanding-Insulin-on-Board-2016-1.pdf

It's says that only corrective blouses are taken in count for IOB on omnipod, im not sure but it can be the explanation

**Comment by Kenoubi** [comment]
@adanvdo I agree with you, this is not working properly.  As far as I can tell, the "Insulin duration in hours" field is totally ignored, as are "Choose basal insulin profile" and "Choose bolus insulin profile. Also used for pump and pen".  But which insulin profiles are enabled and which one you choose when entering a treatment are _not_ ignored; you get different activity curves for FIASP and Humalog, e.g.

The curve it does use for Humalog seems a bit odd, though.  It has an onset of 10 minutes (seems reasonable), a peak at 75 minutes (seems reasonable) and a total duration of 180 minutes (isn't that way too short?)  See https://github.com/NightscoutFoundation/xDrip/blob/master/app/src/main/res/raw/insulin_profiles.txt#L43

For me, Humalog actually seems to peak weirdly fast so I find lying to xDrip+ that I'm using FIASP seems to give more accurate results.  But I wish xDrip+ would just let the user set the curve parameters directly.  Doesn't make sense to me to expect the shape and duration of the curve to be exactly the same in everyone's body.

**Comment by Navid200** [comment]
@gruoner 
Is this a problem, or is it expected behavior?

**Comment by gruoner** [comment]
i didn't implement the humalog-curve because i don't use humalog. When someone has a "better" curve, we can define different onset, peak and finished timestamps.
BUT when using "multiple insulin profile" the "low level prediction values" "insulin duration in hours" is ignored (maybe we can disable this menu item when in multi-insulin).

Because i implemented much more features to multi-insulin as there is currently in the master branch, i can't implement a dialog for changing parameters for trapezoid curves without bothering u with much more new multi-* features.....

**Comment by Navid200** [comment]
@TheConen 
Is this an issue or is it expected behavior?
Thanks

**Comment by TheConen** [comment]
I don't know - the values of Humalog have not been changed with #3157 . However, as we already discussed in #3306, the values that xDrip+ uses for some insulins might very well be wrong and could need some adjusting.
